saleor/graphql/order/mutations/draft_order_create.py

```diff
@@ -101,7 +102,13 @@ class DraftOrderInput(BaseInputObjectType):
         description="ID of a selected shipping method.", name="shippingMethod"
     )
     voucher = graphene.ID(
-        description="ID of the voucher associated with the order.", name="voucher"
+        description="ID of the voucher associated with the order.",
+        name="voucher",
+        deprecation_reason=f"{DEPRECATED_IN_3X_FIELD} Use `voucherCode` instead.",
+    )
+    voucher_code = graphene.ID(
+        description="ID of the voucher code associated with the order.",
+        name="voucherCode",
```

Comment:
In the mutation we need the validation to raise an error when both `voucher` and `voucherCode` are provided

Reply:
I have added validation for this case